ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -35,19 +36,42 @@
       OUTPUT_TABLE, OUTPUT_TOPIC, OUTPUT_TOPIC_EXECUTOR_THREADS, MAX_OUTSTANDING_ELEMENT_COUNT,
       MAX_OUTSTANDING_REQUEST_BYTES);
 
+  @VisibleForTesting
+  protected static class Output implements Function<PubsubMessage, CompletableFuture<Void>> {
```

Comment:
this was added to simplify separating `getOutput` from `getInput` so that `BoundedSink` could inject a termination condition.